### DISCUSSION TITLE
Introduce single alternative ?:

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/csharplang#5942] [Language Ideas]
Is it possible to change C# language in a new version to support a = b ? c which would be equivalent to a = b ? c : a

This would save repeating : a.

**Comment by CyrusNajmabadi** [comment]
`if (b) a = c`

**Reply by ghost** [reply]
Really? Did you even try to understand my point or you prefer thinking that I ask this because I know nothing about C#?

**Reply by ghost** [reply]
My question is simply

Is it possible to change C# to support a = b ? c which would be equivalent to a = b ? c : a;

**Reply by 333fred** [reply]
Is it possible? _Maybe_. Changes to the grammar involving `?` tokens are extremely fraught with potential parsing ambiguities, and I expect this is no different.

**Comment by HaloFour** [comment]
You have to have the else part because the expression is independent of the assignment.  The expression `b ? c` is incomplete because it cannot be evaluated if `b` is not true, but the inability to evaluate the expression is not the same making the assignment conditional.  If you want the assignment itself to be conditional then you should use `if` as suggested by Cyrus.

**Reply by ghost** [reply]
Really? Have I asked this question

Is it possible to use conditional ?: without else part? No I did not. So why are you answering the question I did not ask?

**Reply by ghost** [reply]
If you would write a = b ? c ; that would mean leave a intact in case it is not b. It would be equivalent to a = b ? c : a;

**Reply by HaloFour** [reply]
That is not how assignments or expressions work, so no, it's extremely unlikely that C# would add a feature like that.
